core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -111,9 +113,22 @@ class TaskMetrics private[spark] () extends Serializable {
    * joins. The value of this accumulator should be approximately the sum of the peak sizes
    * across all such data structures created in this task. For SQL jobs, this only tracks all
    * unsafe operators and ExternalSort.
+   * This is not equal to peakOnHeapExecutionMemory + peakOffHeapExecutionMemory
    */
+  // TODO: SPARK-48789: the naming is confusing since this does not really reflect the whole
+  //  execution memory. We'd better deprecate this once we have a replacement.
   def peakExecutionMemory: Long = _peakExecutionMemory.sum
```

Comment:
How about we change its implementation to be `peakOnHeapExecutionMemory + peakOffHeapExecutionMemory`? The current implementation doesn't make much sense due to https://github.com/apache/spark/pull/47192/files#r1692144786

Reply:
Yes I think we can plan this breaking change at spark 4.0

Reply:
peakOnHeapExecutionMemory + peakOffHeapExecutionMemory can peak at different times, so we can't replace it

Reply:
Oh I see, makes sense. Let's leave it then.